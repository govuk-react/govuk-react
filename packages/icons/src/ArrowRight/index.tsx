import * as React from 'react';

import SVG, { SVGProps } from '../SVGBase/index';

interface ArrowRightProps extends SVGProps {
  title?: string;
  fill?: string;
}

const ArrowRight: React.FC<ArrowRightProps> = ({ fill, title, ...rest }: ArrowRightProps) => (
  <SVG viewBox="0 0 17 14" title={title} fill={fill} {...rest}>
    <path d="m10.107-0.0078125-1.4136 1.414 4.2926 4.293h-12.986v2h12.896l-4.1855 3.9766 1.377 1.4492 6.7441-6.4062-6.7246-6.7266z" />
  </SVG>
);

ArrowRight.defaultProps = {
  title: 'arrow right',
  fill: 'currentColor',
};

export default ArrowRight;
