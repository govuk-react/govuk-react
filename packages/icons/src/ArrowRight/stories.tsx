import React from 'react';

import { withKnobs, color, text } from '@storybook/addon-knobs';

import ArrowRightComponent from '.';

export default {
  title: 'Utility/Icons',
  decorators: [withKnobs],
};

export const ArrowRight = () => {
  const fillColour = color('fill', 'black');
  const width = text('width', '256px');
  const height = text('height', '256px');
  return (
    <ArrowRightComponent width={width} height={height} fill={fillColour} title="Example ArrowRight implementation" />
  );
};
