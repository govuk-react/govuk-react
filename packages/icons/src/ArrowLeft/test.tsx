import * as React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import ArrowRight from '.';

const wrapper = <ArrowRight>Example</ArrowRight>;

describe('ArrowRight', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(wrapper, div);
  });

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });
});
