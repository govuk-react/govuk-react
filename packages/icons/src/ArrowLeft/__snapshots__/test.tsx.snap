// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowRight matches wrapper snapshot: wrapper mount 1`] = `
.c0 {
  display: block;
}

<ArrowLeft
  fill="currentColor"
  title="arrow left"
>
  <SVG
    fill="currentColor"
    title="arrow left"
    viewBox="-0.2 0 17 14"
    width="100%"
  >
    <SVGBase__StyledSvg
      fill="currentColor"
      height="100%"
      version="1.1"
      viewBox="-0.2 0 17 14"
      width="100%"
    >
      <svg
        className="c0"
        fill="currentColor"
        height="100%"
        version="1.1"
        viewBox="-0.2 0 17 14"
        width="100%"
      >
        <title>
          arrow left
        </title>
        <path
          d="m6.5938-0.0078125-6.7266 6.7266 6.7441 6.4062 1.377-1.449-4.1856-3.9768h12.896v-2h-12.984l4.2931-4.293-1.414-1.414z"
        />
      </svg>
    </SVGBase__StyledSvg>
  </SVG>
</ArrowLeft>
`;
